src/Confluent.Kafka/Producer.cs

```diff
@@ -381,13 +380,16 @@ public long OutQueueLength
 
         /// <summary>
         ///     Wait until all outstanding produce requests, et.al, are completed. This should typically be done prior
-        //      to destroying a producer instance to make sure all queued and in-flight produce requests are completed
-        //      before terminating.
+        ///     to destroying a producer instance to make sure all queued and in-flight produce requests are completed
+        ///     before terminating. The wait is limited by the optional millisecondsTimeout parameter.
         /// </summary>
-        public void Flush(TimeSpan timeout)
```

Comment:
We discussed that at some point and I think we came to  the conclusion that we should favour  API frugality at this point and add stuff on-demand.
In this particular case it will mostly be called with a constant hardcoded timeout so people will most likely not use TimeSpans, and if they do it is trivial enough to convert  to int